src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+#### Caveats {/*caveats*/}
+
+* The parent function of the `use` Hook must be a React component or your own Hooks.
+* `use` is not recommended for data fetching from a [Server Component](/reference/react/components#server-components). `async` and `await` are preferred when fetching data from a Server Component.
+* When passing promises to `use`, you must store the promise in a state variable or context to avoid recreating the promise on every rerender. When invoking a function that returns a promise, you can to store the returned promise with a cache mechanism like React's `cache` function.
+
+---
+
+## Usage {/*usage*/}
+
+### Reading promises with `use` {/*reading-promises-with-use*/}
+
+`use` functions similarly to the `await` with JavaScript promises but is compatible with React's renderer and [`Suspense`](/reference/react/Suspense). Browser APIs or data fetching libraries that return a promise can be passed to `use` to read the value of the promise.
+
+In the example below we'll use <CodeStep step={1}>[`navigator.storage.estimate()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate)</CodeStep>, a browser API that provides information on how much storage is being used by the current page and how much space is available. <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is an async function that returns a Promise which resolves once the information is available.
+
+```js [[1,6, "navigator.storage.estimate()"], [2,6, "storagePromise"], [3,6, "useState"], [4,7, "use"], [5,8, "storage.quota"]]
+"use client";
+
+import { use, useState } from "react"
+
+function StorageInfo() {
+  const [storagePromise] = useState(navigator.storage.estimate());
+  const storage = use(storagePromise);
+  return <p>An estimated {storage.quota} bytes available</p>
+}
+```
+
+1. First, the promise returned by <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is stored in the <CodeStep step={2}>`storagePromise`</CodeStep> state variable created by <CodeStep step={3}>`useState`</CodeStep>. This prevents the call to <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> and the promise it returns from being recreated across renders.
+2. Next, <CodeStep step={2}>`storagePromise`</CodeStep> is passed to <CodeStep step={4}>`use`</CodeStep> to read the resolved value of the promise.
+3. Finally, the resolved value of the promise can now be used safely. This includes surfacing data read by <CodeStep step={4}>`use`</CodeStep> like <CodeStep step={5}>`storage.quota`</CodeStep> in the JSX returned by the component.
+
+<Pitfall>
+
+When passing promises to `use`, the promise must be cached with the `cache` function, a state variable, or context to prevent the promise from being recreated on every render.
```

Comment:
We mention `cache` here for the first time, but it's not yet available in the docs. That might be okay, but I don't know if that creates a dependency on shipping that first.

Reply:
@lunaleaps is working on the `cache` doc now, I think its ok to ship this as is and add a link once the `cache` doc is publish. Does that sound ok to you?

Reply:
Works for me 👍